prow/flagutil/github.go

```diff
@@ -114,30 +119,48 @@ func (o *GitHubOptions) GitHubClient(secretAgent *secret.Agent, dryRun bool) (cl
 
 // GitClient returns a Git client.
 func (o *GitHubOptions) GitClient(secretAgent *secret.Agent, dryRun bool) (client *git.Client, err error) {
-	client, err = git.NewClient()
+	old, _, err := o.GitClients(secretAgent, dryRun)
+	return old, err
+}
+
+// GitClients returns both old and new Git clients.
+func (o *GitHubOptions) GitClients(secretAgent *secret.Agent, dryRun bool) (old *git.Client, new git.Clients, err error) {
```

Comment:
Why are they `old and new`? Thy serve different use cases, we very much need the `old` one still or not?

Reply:
For now, yes, in the future, no

Reply:
So you plan to eventually add caching to this client? How is that supposed to look like?

Reply:
I will re-use the repo client but generate the underlying filesystem as per the other client. This just has interfaces that let us inject dependencies for testing, unlike the other client.